pkg/controller/deployment/sync.go

```diff
@@ -503,15 +503,26 @@ func (dc *DeploymentController) updateDeploymentStatus(allRSs []*extensions.Repl
 
 // isScalingEvent checks whether the provided deployment has been updated with a scaling event
 // by looking at the desired-replicas annotation in the active replica sets of the deployment.
-func (dc *DeploymentController) isScalingEvent(d *extensions.Deployment) bool {
+func (dc *DeploymentController) isScalingEvent(d *extensions.Deployment) (bool, error) {
 	newRS, oldRSs, err := dc.getAllReplicaSetsAndSyncRevision(d, false)
 	if err != nil {
-		return false
+		return false, err
 	}
 	// If there is no new replica set matching this deployment and the deployment isn't paused
 	// then there is a new rollout that waits to happen
 	if newRS == nil && !d.Spec.Paused {
-		return false
+		// Update all active replicas sets to the new deployment size. SetReplicasAnnotations makes
+		// sure that we will update only replica sets that don't have the current size of the deployment.
+		maxSurge := deploymentutil.MaxSurge(*d)
+		for _, rs := range controller.FilterActiveReplicaSets(oldRSs) {
+			if updated := deploymentutil.SetReplicasAnnotations(rs, d.Spec.Replicas, d.Spec.Replicas+maxSurge); updated {
```

Comment:
It'd be more robust if we don't calculate desiredReplicas & maxReplicas here but inside `SetReplicasAnnotations` based one deployment spec. WDYT?